update-index.nix

```diff
@@ -0,0 +1,28 @@
+{
+  pkgs ? import <nixpkgs> {},
+
+  coreutils ? pkgs.coreutils,
+  gnugrep ? pkgs.gnugrep,
+  lib ? pkgs.lib,
+  nix-index ? pkgs.nix-index,
+  writeScript ? pkgs.writeScript,
+}:
+
+writeScript "update-index" ''
+  PATH=${lib.makeBinPath [
+    coreutils
+    gnugrep
+    nix-index
+  ]}
+
+  # on flake based installations nixpkgs is specified via
+  # flake input and therefore NIX_PATH might be unset
+  if echo $NIX_PATH | grep -q "nixpkgs="; then
+    nixpkgs=""
+  else
+    nixpkgs="-I nixpkgs=${pkgs.path}"
+  fi
+
+  mkdir -p $HOME/.cache/comma/
+  nix-index -d $HOME/.cache/comma/index -f $nixpkgs
```

Comment:
Why is this not re-using the db from nix-index? Having the DB twice is not bad for disk usage as it is only 22MB but downloading takes a significant amount of network traffic and is quite resource heavy. Also we should not waste resources from the nixos cache.

Reply:
A database that only contains binaries from `$out/bin` would be useful here and should significantly smaller than the full index... I would also build it here if it was available: https://github.com/Mic92/nix-index-database

Reply:
We can maybe hijack the nixpkgs argument here https://github.com/bennofs/nix-index/blob/master/src/nixpkgs.rs#L38 to only list bin things.

Reply:
definite +1 for this - comma doesn't cause network issues for me, but this would still be an irritating time sink especially when perfectly good solutions already exist to build the nix-index db in a less disruptive fashion.

Reply:
I changed the DB paths to the default location now to share the DC with the nix-index cli.